### ISSUE TITLE
missing features / questions

### ISSUE THREAD

**Issue by thezoggy** [impredicative/irc-rss-feed-bot#65]
I have been using rss-synd (eggdrop tcl script) for ages but was looking to move away from tcl.
I stumbled upon this bot, and while it looks like there is lots of features I am not seeing some things and figured I'd post here to check.

* Is there SASL support for bot authentication to irc server
* Ability to post the same feed to multiple channels.
* Ability to call command in channel to re-post last x items

* Ability to set specifically how many items are shared on an update
I see the `new` config option to set how many items on the feed are share - any plans to make this more flexible, for example if I wanted to do 5 items (without having to modify the sourcecode)

**Comment by impredicative** [comment]
> Is there SASL support for bot authentication to irc server

Yes, the underlying package [`miniirc`](https://github.com/luk3yx/miniirc) does support SASL. I use the bot on Freenode with automatic SASL.

> Ability to post the same feed to multiple channels.

Yes, but the feeds will post on independent schedules.

> Ability to call command in channel to re-post last x items

No, and there is no plan to ever implement it. Having said this, improved search capabilities, including to link to search results of the last *n* items for a channel, are on the medium-term roadmap, with various possibilities for implementation.

> Ability to set specifically how many items are shared on an update. I see the new config option to set how many items on the feed are share - any plans to make this more flexible, for example if I wanted to do 5 items (without having to modify the sourcecode)

No, and there is no plan to ever implement it. One reason among many is that there is serious potential for entries to be lost if this feature is implemented. As for the `new: some` feature, it discards all posts except three, but just the first time around for a feed.

---

In summary, if I had to write the bot again, I'd make its various components a lot more decoupled. Currently there is too much coupling between its components.

**Comment by thezoggy** [comment]
How do i do multiple channels with the same feed? Or do i just have to create multiple entries (duplicating the feed with different channels) ?

**Comment by impredicative** [comment]
> How do i do multiple channels with the same feed? Or do i just have to create multiple entries (duplicating the feed with different channels) ?

I don't know why this is difficult. Yes, you've to create an entry for the feed under each channel. You can optionally choose to use YAML anchors to avoid significant duplication.

**Comment by thezoggy** [comment]
Not difficult, just was confirming as it is different than other apps where they do one feed url to many channels while I could not tell if yours supported the same from the docs.

**Comment by impredicative** [comment]
Even so, the current config layout works. Feed URLs that are used in multiple channels are often used with channel-specific topic filters anyway.

Again, I encourage the use of YAML anchors.

**Comment by thezoggy** [comment]
is there a way to reload the config without stopping/starting the bot? (i did not see admin command to do so)

**Comment by impredicative** [comment]
> is there a way to reload the config without stopping/starting the bot? (i did not see admin command to do so)

No. This is not easy to implement with the current threading architecture. It will be very difficult to fully implement even with a revised architecture.
